src/app/components/fees-row/fees-row.tsx

```diff
@@ -0,0 +1,88 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { StackProps } from '@stacks/ui';
+import BigNumber from 'bignumber.js';
+import { useField } from 'formik';
+
+import { CryptoCurrencies } from '@shared/models/currencies.model';
+import { FeeTypes, Fees } from '@shared/models/fees/_fees.model';
+import { createMoney } from '@shared/models/money.model';
+import { isNumber, isString } from '@shared/utils';
+
+import { useConvertCryptoCurrencyToFiatAmount } from '@app/common/hooks/use-convert-to-fiat-amount';
+import { convertAmountToBaseUnit } from '@app/common/money/calculate-money';
+import { LoadingRectangle } from '@app/components/loading-rectangle';
+
+import { FeesRowLayout } from './components/fees-row.layout';
+
+interface FeeRowProps extends StackProps {
+  fees?: Fees;
+  isSponsored: boolean;
+}
+export function FeesRow(props: FeeRowProps): JSX.Element {
+  const { fees, isSponsored, ...rest } = props;
+  const [field, _, helper] = useField('fee');
+  const [, __, feeTypeHelper] = useField('feeType');
+  const [fieldWarning, setFieldWarning] = useState<string | undefined>(undefined);
+  const [selected, setSelected] = useState(FeeTypes.Middle);
+  const [isCustom, setIsCustom] = useState(false);
+  const [isSelectVisible, setIsSelectVisible] = useState(false);
+
+  const hasFeeEstimates = fees?.estimates.length;
+  const feeCurrencySymbol = fees?.estimates[0].fee.symbol as CryptoCurrencies;
+  const convertCryptoCurrencyToUsd = useConvertCryptoCurrencyToFiatAmount(feeCurrencySymbol);
+
+  const feeInUsd = useMemo(() => {
+    if ((!isNumber(field.value) && !isString(field.value)) || !feeCurrencySymbol) return null;
+    const feeAsMoney = createMoney(new BigNumber(field.value), feeCurrencySymbol);
+    return convertCryptoCurrencyToUsd(feeAsMoney);
+  }, [convertCryptoCurrencyToUsd, feeCurrencySymbol, field.value]);
+
+  useEffect(() => {
+    if (hasFeeEstimates && !field.value && !isCustom) {
+      helper.setValue(convertAmountToBaseUnit(fees.estimates[FeeTypes.Middle].fee).toNumber());
+      feeTypeHelper.setValue(FeeTypes[FeeTypes.Middle]);
+    }
+    if (isSponsored) {
+      helper.setValue(0);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [fees?.estimates, isCustom]);
```

Comment:
What's the benefit of not having all deps?

Reply:
If I remem correctly, we only want this to run when the fee estimate array loads.